internal/dag/dag.go

```diff
@@ -59,10 +59,13 @@ type DAG struct {
 	// StatusCache holds a cache of status updates to send.
 	StatusCache status.Cache
 
-	Listeners          []*Listener
-	VirtualHosts       map[string]*VirtualHost
-	SecureVirtualHosts map[string]*SecureVirtualHost
-	ExtensionClusters  []*ExtensionCluster
+	Listeners         map[string]*Listener
+	ExtensionClusters []*ExtensionCluster
+
+	// Set this to true if Contour is configured with a Gateway
+	// and Listeners are derived from the Gateway's Listeners, or
+	// false otherwise.
+	HasNonStaticListeners bool
```

Comment:
tiny naming nit: could use a name w/o negation to make situations like `!HasNonStaticListeners` read a little better (e.g. `HasDynamicListeners`)